File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = 1
        maxprofit = 0
        
        while (right < len(prices)):
            if prices[right] < prices[left]:
                left = right
            elif prices[right] - prices[left] > maxprofit:
                maxprofit = prices[right] - prices[left]
            right += 1
                
        return maxprofit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int left = 0;
        int right = 1;
        int maxP = 0;
        while(left < right && right < n) {
            if(prices[right] - prices[left] <= 0) {
                left = right;
                right++;
            }
            else if (prices[right] - prices[left] > maxP) {
                maxP = prices[right] - prices[left];
                right++;
            }
            else {
                right++;
            }
        }
        return maxP;
    }
}